addons/osfstorage/views.py

```diff
@@ -282,6 +282,8 @@ def osfstorage_get_children(file_node, **kwargs):
 def osfstorage_create_child(file_node, payload, **kwargs):
```

Comment:
This is where I need input - the main point of this work is to be able to support uploading a new version of a file that's already in this location.  

For background, we decide that it's a new version by attempting to append a new file in this location with the same name. If we get an integrityerror (a file with this name already exists in this location), then we attempt to create a new version of that file instead.

The new behavior, then, needs to support uploading a new version that is an upload + a rename, essentially.  So I'm thinking the payload still has `name` which is the original name of the file, so we can locate the file for which we want to create a new version.  But if `new_name` exists in the payload, this signals that we want this latest version to also have a new name.  The file, plus the fileversion in the thru table will also get a new name.

Reply:
not going this direction any longer.